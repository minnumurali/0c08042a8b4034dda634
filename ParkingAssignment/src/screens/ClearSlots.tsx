import React, { useMemo } from "react";
import { View, FlatList, StyleSheet, Alert } from 'react-native';
import { Paragraph, Title, List, Button } from "react-native-paper";
import moment from "moment";

import { checkStatus } from "../actions";
import { useAppState } from "../hooks/useAppState";
import { useDispatch } from "../hooks/useDispatch";
import { ScreenNavigation } from "../types/navigation";
import { Slot } from '../types/context';
import { UPDATE_SLOTS } from '../contexts/actions';
import { Moment } from 'moment';

type Props = {
    navigation: ScreenNavigation;
}

const ClearSlots: React.FC<Props> = ({ navigation }) => {

    const state = useAppState();
    const dispatch = useDispatch();

    const slotsArray = useMemo(() => {
        return Object.values(state.slots);
    }, [state]);

    const calculateAmount = (time: Moment): number => {
        const hours = moment().diff(time, "hours");
        if(hours <= 2){
            return 10;
        }
        return (hours - 1) * 10;
    }

    const calculateTime = (time: Moment): number => {
        return moment().diff(time, "hours") || 1;
    }

    const clearSlot = async (id: string) => {
        const { [id]: slot, ...rest } = state.slots;
        const resp = await checkStatus(slot.regNo, calculateAmount(slot.createdAt));
        if(resp){
            dispatch({
                type: UPDATE_SLOTS,
                payload: {
                    ...rest,
                    [id]: {
                        id: id,
                        regNo: "",
                        createdAt: moment()
                    }
                }
            });
        } else {
            Alert.alert("Error", "Status check failed");
        }
    }

    return(
        <View
            style={styles.container}
            testID="deregister-car-registration"
        >
            <FlatList
                data={slotsArray}
                keyExtractor={slot => slot.id}
                ListEmptyComponent={<Title>No booked slots...</Title>}
                renderItem={({ item, index }) => {
                    return(
                        <List.Section
                            style={styles.section}
                            testID={item.regNo? `parking-drawing-registered-${index}`: `parking-drawing-space-${index}`}
                        >
                            <Title testID={`parking-drawing-space-number-${index}`}>ID: {item.id}</Title>
                            <Paragraph>Reg No: 
                                <Paragraph style={{ color: item.regNo? "red": "green" }}>{item.regNo || "Free slot"}</Paragraph>
                            </Paragraph>
                            <Paragraph testID="deregister-time-spent">Time spent: {item.regNo? calculateTime(item.createdAt): 0} hr</Paragraph>
                            <Paragraph testID="deregister-charge">Amount: {item.regNo? calculateAmount(item.createdAt): 0}</Paragraph>
                            <Button
                                onPress={() => clearSlot(item.id)}
                                contentStyle={styles.button}
                                disabled={!item.regNo}
                                testID="deregister-payment-button"
                            >
                                Payment Taken
                            </Button>
                        </List.Section>
                    );
                }}
            />
            <Button
                onPress={navigation.goBack}
                mode="contained"
                contentStyle={styles.button}
                testID="deregister-back-button"
            >
                Back
            </Button>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        padding: "5%"
    },
    button: {
        height: 50
    },
    section: {
        borderColor: "#cccccc",
        borderBottomWidth: 1
    }
});

export default ClearSlots;