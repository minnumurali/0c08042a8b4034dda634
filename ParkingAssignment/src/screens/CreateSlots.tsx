import React, { useState, useMemo } from "react";
import { View, StyleSheet } from "react-native";
import { TextInput, Title, Button } from "react-native-paper";
import moment from "moment";

import { useAppState } from "../hooks/useAppState";
import { useDispatch } from "../hooks/useDispatch";
import { ScreenNavigation } from "../types/navigation";
import { ParkingLot, Slot } from '../types/context';
import { CREATE_SLOTS } from '../contexts/actions';

type Props = {
    navigation: ScreenNavigation;
}

const CreateSlots: React.FC<Props> = ({ navigation }) => {

    const [slots, setSlots] = useState<string>("");
    const state = useAppState();
    const dispatch = useDispatch();

    const slotCount = useMemo(() => {
        return Object.keys(state.slots).length
    }, [state]);

    const createSlots = () => {
        const count = Number(slots);
        if(count){
            const slotMap: ParkingLot = {};
            for(let i = 0; i < count; i++){
                const slotId: string = Math.random().toString();
                const slot: Slot = {
                    id: slotId,
                    createdAt: moment(),
                    regNo: ""
                }
                slotMap[slotId] = slot;
            }
            dispatch({ type: CREATE_SLOTS, payload: slotMap });
            setSlots("");
            navigation.goBack();
        }
    }

    return(
        <View style={styles.container}>
            <Title>{slotCount? slotCount: "No"} slots are created so far</Title>
            <TextInput
                value={slots}
                onChangeText={setSlots}
                mode="outlined"
                keyboardType="number-pad"
                maxLength={2}
                style={styles.input}
                testID="parking-create-text-input"
            />
            <Button
                mode="contained"
                onPress={createSlots}
                disabled={!slots}
                contentStyle={styles.button}
                testID="parking-create-submit-button"
            >
                Create
            </Button>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        padding: "5%"
    },
    input: {
        marginBottom: 20
    },
    button: {
        height: 50
    }
});

export default CreateSlots;