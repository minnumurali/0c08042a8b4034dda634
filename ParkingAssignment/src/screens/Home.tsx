import React from "react";
import { View, StyleSheet } from "react-native";
import { TouchableRipple, List } from "react-native-paper";

import { ScreenNavigation } from "../types/navigation";

type Props = {
    navigation: ScreenNavigation;
}

const Home: React.FC<Props> = ({ navigation }) => {

    const createSlots = () => navigation.navigate("CreateSlots");
    const registerSlots = () => navigation.navigate("RegisterSlots");
    const clearSlots = () => navigation.navigate("ClearSlots");

    return(
        <View style={styles.container} testID="home">
            <List.Section>
                <List.Subheader>Actions</List.Subheader>
                <TouchableRipple onPress={createSlots}>
                    <List.Item title="Create Slots" />
                </TouchableRipple>
                <TouchableRipple onPress={registerSlots}>
                    <List.Item title="Register Slots" />
                </TouchableRipple>
                <TouchableRipple onPress={clearSlots}>
                    <List.Item title="Clear Slots" />
                </TouchableRipple>
            </List.Section>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        padding: "5%"
    }
});

export default Home;