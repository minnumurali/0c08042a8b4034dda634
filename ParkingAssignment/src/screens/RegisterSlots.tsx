import React, { useState } from "react"
import { View, StyleSheet, Alert } from "react-native"
import { TextInput, Button } from "react-native-paper";
import moment from "moment";

import { useAppState } from "../hooks/useAppState";
import { useDispatch } from "../hooks/useDispatch";
import { ScreenNavigation } from "../types/navigation";
import { Slot } from '../types/context';
import { REGISTER_SLOT } from '../contexts/actions';

type Props = {
    navigation: ScreenNavigation;
}

const RegisterSlots: React.FC<Props> = ({ navigation }) => {

    const [regNo, setRegNo] = useState<string>("");
    const state = useAppState();
    const dispatch = useDispatch();

    const registerSlot = () => {
        let doesIdExist: boolean = false;
        const freeSlotId = Object.keys(state.slots).find(slot => {
            if(state.slots[slot].regNo === regNo){
                doesIdExist = true;
            }
            return state.slots[slot].regNo === "";
        });
        if(doesIdExist){
            Alert.alert("", "This vehicle is already registered.");
            return;
        }
        if(!freeSlotId){
            Alert.alert("", "All slots are filled. Please try once slots are available.");
            return;
        }

        const newSlot: Slot = {
            id: freeSlotId,
            createdAt: moment(),
            regNo: regNo
        }
        dispatch({ type: REGISTER_SLOT, payload: newSlot });
        setRegNo("");
        Alert.alert("", "Registered successfully.");
        setTimeout(() => {
            navigation.goBack();
        }, 1000);
    }

    return(
        <View style={styles.container}>
            <TextInput
                value={regNo}
                onChangeText={setRegNo}
                mode="outlined"
                maxLength={10}
                style={styles.input}
                testID="parking-drawing-registration-input"
            />
            <Button
                mode="contained"
                onPress={registerSlot}
                disabled={!regNo}
                contentStyle={styles.button}
                testID="parking-drawing-add-car-button"
            >
                Register
            </Button>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        padding: "5%"
    },
    input: {
        marginBottom: 20
    },
    button: {
        height: 50
    }
});

export default RegisterSlots;