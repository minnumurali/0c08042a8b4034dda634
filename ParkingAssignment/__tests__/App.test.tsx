import { render, fireEvent } from "@testing-library/react-native";
import Home from "../src/screens/Home";
import CreateSlots from '../src/screens/CreateSlots';
import RegisterSlots from '../src/screens/RegisterSlots';
import ClearSlots from '../src/screens/ClearSlots';

describe("App test suite", () => {

    const props = {
        navigation: {
            navigate: jest.fn(),
            goBack: jest.fn()
        }
    }

    it("works", () => {
        const test = true;
        expect(test).toBe(true);
    });

    it("renders home screen", () => {
        render(<Home { ...props } />);
    });

    it("renders CreateSlots screen", () => {
        render(<CreateSlots { ...props } />);
    });

    it("renders RegisterSlots screen", () => {
        render(<RegisterSlots { ...props } />);
    });

    it("renders ClearSLots screen", () => {
        render(<ClearSlots { ...props } />);
    });

    it("matches CreateSlots snapshot", () => {
        const tree = render(<CreateSlots { ...props } />);
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it("matches RegisterSlots screen", () => {
        const tree = render(<RegisterSlots { ...props } />);
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it("renders ClearSlots screen", () => {
        const tree = render(<ClearSlots { ...props } />);
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it("matches home testId", () => {
        const tree = render(<Home />);
        expect(tree.findByTestId("home")).toBeTruthy();
    });

    it("matches CreateSlots testID", () => {
        const tree = render(<CreateSlots />);
        expect(tree.findByTestId("parking-create-submit-button")).toBeTruthy();
        
    });
});