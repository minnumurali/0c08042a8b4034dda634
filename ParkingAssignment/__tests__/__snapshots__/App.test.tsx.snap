// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test suite matches CreateSlots snapshot 1`] = `
<View
  style={
    {
      "flex": 1,
      "padding": "5%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
        },
        {
          "textAlign": "left",
        },
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "System",
            "fontWeight": "500",
            "writingDirection": "ltr",
          },
          [
            {
              "fontSize": 20,
              "letterSpacing": 0.15,
              "lineHeight": 30,
              "marginVertical": 2,
            },
            undefined,
          ],
        ],
      ]
    }
  >
    No
     slots are created so far
  </Text>
  <View
    style={
      {
        "marginBottom": 20,
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 6,
          },
          {
            "backgroundColor": "#f6f6f6",
            "borderColor": "rgba(0, 0, 0, 0.54)",
            "borderRadius": 4,
            "borderWidth": 1,
          },
        ]
      }
      testID="text-input-outline"
    />
    <View>
      <View
        style={
          [
            {
              "paddingBottom": 0,
            },
            {
              "minHeight": 64,
              "paddingTop": 8,
            },
          ]
        }
      >
        <TextInput
          editable={true}
          keyboardType="number-pad"
          maxFontSizeMultiplier={1.5}
          maxLength={2}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholderTextColor="rgba(0, 0, 0, 0.54)"
          selectionColor="#6200ee"
          style={
            [
              {
                "flexGrow": 1,
                "margin": 0,
                "paddingHorizontal": 14,
                "zIndex": 1,
              },
              {
                "height": 56,
              },
              {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontVariant": undefined,
                "fontWeight": undefined,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              [
                {},
              ],
            ]
          }
          testID="parking-create-text-input"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgba(0, 0, 0, 0.12)",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 0,
        "minWidth": 64,
        "shadowColor": "#000000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "borderRadius": 4,
          },
        ]
      }
      testID="parking-create-submit-button"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            {
              "height": 50,
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              [
                {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                {
                  "textTransform": "uppercase",
                },
                {
                  "color": "rgba(0, 0, 0, 0.32)",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Create
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`App test suite matches RegisterSlots screen 1`] = `
<View
  style={
    {
      "flex": 1,
      "padding": "5%",
    }
  }
>
  <View
    style={
      {
        "marginBottom": 20,
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 6,
          },
          {
            "backgroundColor": "#f6f6f6",
            "borderColor": "rgba(0, 0, 0, 0.54)",
            "borderRadius": 4,
            "borderWidth": 1,
          },
        ]
      }
      testID="text-input-outline"
    />
    <View>
      <View
        style={
          [
            {
              "paddingBottom": 0,
            },
            {
              "minHeight": 64,
              "paddingTop": 8,
            },
          ]
        }
      >
        <TextInput
          editable={true}
          maxFontSizeMultiplier={1.5}
          maxLength={10}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholderTextColor="rgba(0, 0, 0, 0.54)"
          selectionColor="#6200ee"
          style={
            [
              {
                "flexGrow": 1,
                "margin": 0,
                "paddingHorizontal": 14,
                "zIndex": 1,
              },
              {
                "height": 56,
              },
              {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontVariant": undefined,
                "fontWeight": undefined,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              [
                {},
              ],
            ]
          }
          testID="parking-drawing-registration-input"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgba(0, 0, 0, 0.12)",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 0,
        "minWidth": 64,
        "shadowColor": "#000000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "borderRadius": 4,
          },
        ]
      }
      testID="parking-drawing-add-car-button"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            {
              "height": 50,
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              [
                {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                {
                  "textTransform": "uppercase",
                },
                {
                  "color": "rgba(0, 0, 0, 0.32)",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Register
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`App test suite matches home snapshot 1`] = `
<View
  style={
    {
      "flex": 1,
      "padding": "5%",
    }
  }
>
  <View
    style={
      [
        {
          "marginVertical": 8,
        },
        undefined,
      ]
    }
    theme={
      {
        "animation": {
          "scale": 1,
        },
        "colors": {
          "accent": "#03dac4",
          "backdrop": "rgba(0, 0, 0, 0.5)",
          "background": "#f6f6f6",
          "disabled": "rgba(0, 0, 0, 0.26)",
          "error": "#B00020",
          "notification": "#f50057",
          "onSurface": "#000000",
          "placeholder": "rgba(0, 0, 0, 0.54)",
          "primary": "#6200ee",
          "surface": "#ffffff",
          "text": "#000000",
        },
        "dark": false,
        "fonts": {
          "light": {
            "fontFamily": "System",
            "fontWeight": "300",
          },
          "medium": {
            "fontFamily": "System",
            "fontWeight": "500",
          },
          "regular": {
            "fontFamily": "System",
            "fontWeight": "400",
          },
          "thin": {
            "fontFamily": "System",
            "fontWeight": "100",
          },
        },
        "roundness": 4,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          {
            "textAlign": "left",
          },
          [
            {
              "paddingHorizontal": 16,
              "paddingVertical": 13,
            },
            {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontFamily": "System",
              "fontWeight": "500",
            },
            undefined,
          ],
        ]
      }
    >
      Actions
    </Text>
    <View
      accessibilityState={
        {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          false,
          undefined,
        ]
      }
    >
      <View
        accessibilityState={
          {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            false,
            [
              {
                "padding": 8,
              },
              undefined,
            ],
          ]
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              [
                {
                  "marginVertical": 6,
                  "paddingLeft": 8,
                },
                {
                  "flex": 1,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  {
                    "textAlign": "left",
                  },
                  [
                    {
                      "fontSize": 16,
                    },
                    {
                      "color": "rgba(0, 0, 0, 0.87)",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Create Slots
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          false,
          undefined,
        ]
      }
    >
      <View
        accessibilityState={
          {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            false,
            [
              {
                "padding": 8,
              },
              undefined,
            ],
          ]
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              [
                {
                  "marginVertical": 6,
                  "paddingLeft": 8,
                },
                {
                  "flex": 1,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  {
                    "textAlign": "left",
                  },
                  [
                    {
                      "fontSize": 16,
                    },
                    {
                      "color": "rgba(0, 0, 0, 0.87)",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Register Slots
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          false,
          undefined,
        ]
      }
    >
      <View
        accessibilityState={
          {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            false,
            [
              {
                "padding": 8,
              },
              undefined,
            ],
          ]
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              [
                {
                  "marginVertical": 6,
                  "paddingLeft": 8,
                },
                {
                  "flex": 1,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  {
                    "textAlign": "left",
                  },
                  [
                    {
                      "fontSize": 16,
                    },
                    {
                      "color": "rgba(0, 0, 0, 0.87)",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Clear Slots
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`App test suite renders ClearSlots screen 1`] = `
<View
  style={
    {
      "flex": 1,
      "padding": "5%",
    }
  }
  testID="deregister-car-registration"
>
  <RCTScrollView
    ListEmptyComponent={
      <Title>
        No booked slots...
      </Title>
    }
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <Text
        onLayout={[Function]}
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "500",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontSize": 20,
                  "letterSpacing": 0.15,
                  "lineHeight": 30,
                  "marginVertical": 2,
                },
                undefined,
              ],
            ],
          ]
        }
      >
        No booked slots...
      </Text>
    </View>
  </RCTScrollView>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#6200ee",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 2,
        "minWidth": 64,
        "shadowColor": "#000000",
        "shadowOffset": {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "borderRadius": 4,
          },
        ]
      }
      testID="deregister-back-button"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            {
              "height": 50,
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              [
                {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                {
                  "textTransform": "uppercase",
                },
                {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Back
        </Text>
      </View>
    </View>
  </View>
</View>
`;
